sentry-core/src/main/java/io/sentry/core/cache/SessionCache.java

```diff
@@ -153,9 +153,12 @@ public void store(final @NotNull SentryEnvelope envelope, final @Nullable Object
         } catch (IOException e) {
           options.getLogger().log(SentryLevel.ERROR, "Error processing session.", e);
         }
-      } else {
-        updateCurrentSession(currentSessionFile, envelope);
+
+        if (!currentSessionFile.delete()) {
```

Comment:
at this point the leftover session and its current session file already became a new envelope file to be sent, so deleting it as the new session will take place.